src/dataflow/src/render/sources.rs

```diff
@@ -503,11 +506,44 @@ where
                                         source_persist_config,
                                     ) = source_persist_config
                                     {
+                                        let bindings_handle = source_persist_config
+                                            .bindings_config
+                                            .write_handle
+                                            .clone();
+                                        let upsert_state_handle =
+                                            source_persist_config.upsert_config.write_handle;
+
                                         let sealed_upsert = upsert_ok.conditional_seal(
                                             &source_name,
                                             &ts_bindings,
-                                            source_persist_config.upsert_config.write_handle,
-                                            source_persist_config.bindings_config.write_handle,
+                                            upsert_state_handle.clone(),
+                                            bindings_handle.clone(),
+                                        );
+
+                                        let allowed_compaction_frontier = Rc::new(RefCell::new(
+                                            Antichain::<Timestamp>::from_elem(
+                                                TimelyTimestamp::minimum(),
+                                            ),
+                                        ));
+                                        render_state.allowed_compaction_frontiers.insert(
```

Comment:
as written this will overwrite the compaction frontier if someone creates two instances of the same source - not sure if we want to check if `orig_id` already exists in the hashmap / debug assert that the return value of insert is none. Perhaps an alternate answer is "multiple instances of the same source are borked anyway, so nothing better to do for now", which is fine but then maybe we should have a todo?

Reply:
I'm adding an assert. But yes, there is an open issue about preventing multiple rendered instances for the same persistent source.